helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -210,6 +213,18 @@ private void updateStateTransitionMessageThreadPool(Message message, HelixManage
       return;
     }
 
+    if (!_batchMessageThreadpoolChecked) {
```

Comment:
Would it be better to pass it in? Each HelixManager for sure has a DefaultMessagingService, which for sure has a HelixTaskExecutor. Using NotificationContext to pass in the "parent object" just to do initialization doesn't seem quite straightforward.

When I work on adding ThreadPoolExecutorMonitor for HelixTaskExecutor, I actually have the same problem of getting InstanceType and ClusterName from HelixManager. Maybe we should add it in the constructor?